src/Libs/Soa/v1/Yousign.php

```diff
@@ -201,4 +208,86 @@ public function activateSignature(string $signatureRequestId): ActivateSignature
 
         return new ActivateSignatureResponse($response->json());
     }
+
+    /**
+     * @param string $signatureRequestId
+     *
+     * @return InitiateSignatureResponse
+     */
+    public function getSignatureById(string $signatureRequestId): InitiateSignatureResponse
+    {
+        /** @var string $url */
+        $url = self::INITIATE_SIGNATURE_URL . self::URL_SEPARATOR . $signatureRequestId;
+
+        /** @var Response $response */
+        $response = $this->apiClient->get($url);
+
+        if (!is_array($response->json())) {
+            throw new RuntimeException('Yousign response is not an array.');
+        }
+
+        return new InitiateSignatureResponse($response->json());
+    }
+
+    /**
+     * @param string $signatureRequestId
+     * @param string $documentId
+     *
+     * @return string
+     */
+    public function getDocumentById(string $signatureRequestId, string $documentId): string
+    {
+        /** @var string $url */
+        $url = self::INITIATE_SIGNATURE_URL . self::URL_SEPARATOR . $signatureRequestId . self::URL_SEPARATOR . self::UPLOAD_DOCUMENT_URL . self::URL_SEPARATOR . $documentId;
+
+        /** @var Response $response */
+        $response = $this->apiClient->get($url);
+
+        if (!is_string($response->body())) {
+            throw new RuntimeException('Yousign response is not a string.');
+        }
+
+        return $response->body();
+    }
+
+    /**
+     * @param string $signatureRequestId
+     * @param string $signerId
+     *
+     * @return string
+     */
+    public function getAuditTrail(string $signatureRequestId, string $signerId): string
+    {
+        /** @var string $url */
+        $url = self::INITIATE_SIGNATURE_URL . self::URL_SEPARATOR . $signatureRequestId . self::URL_SEPARATOR . self::ADD_SIGNER_URL . Soa::URL_SEPARATOR . $signerId . self::URL_SEPARATOR . self::DOWNLOAD_AUDIT_TRAIL;
```

Comment:
Si potrebbe anche strutturare meglio la generazione degli url, ma tutto sommato possiamo tenere anche così, però usa il metodo `implode()` che ti permette di concentrarti meglio sui singoli segmenti che compongono l'url finale, senza questa sintassi così ripetititva.

Reply:
andrebbero cambiati ovunque, vedo come sono messa con i tempi. Magari lo faccio "a tempo perso" più in la